File: app/src/main/java/com/BlizzardArmory/model/warcraft/account/Character_.java

package com.BlizzardArmory.model.warcraft.account;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * The type Character.
 */
public class Character_ {

    @SerializedName("href")
    @Expose
    private String href;

    /**
     * Gets href.
     *
     * @return the href
     */
    public String getHref() {
        return href;
    }

    /**
     * Sets href.
     *
     * @param href the href
     */
    public void setHref(String href) {
        this.href = href;
    }

}

File: app/src/main/java/com/BlizzardArmory/model/warcraft/account/Character.kt
package com.BlizzardArmory.model.warcraft.account

import androidx.annotation.Keep
import com.BlizzardArmory.model.warcraft.common.Faction
import com.BlizzardArmory.model.warcraft.common.Realm
import com.google.gson.annotations.SerializedName


/**
 * The type Character.
 */
@Keep
data class Character(

    @SerializedName("character")
    var characterSelf: CharacterSelf,

    @SerializedName("protected_character")
    var protectedCharacter: ProtectedCharacter,

    @SerializedName("name")
    var name: String,

    @SerializedName("id")
    var id: Long,

    @SerializedName("realm")
    var realm: Realm,

    @SerializedName("playable_class")
    var playableClass: PlayableClass,

    @SerializedName("playable_race")
    var playableRace: PlayableRace,

    @SerializedName("gender")
    var gender: Gender,

    @SerializedName("faction")
    var faction: Faction,

    @SerializedName("level")
    var level: String

)